frontend/packages/@depmap/dataset-manager/src/components/index.tsx

```diff
@@ -36,14 +35,15 @@ export default function Datasets() {
   );
   const [showDatasetModal, setShowDatasetModal] = useState(false);
   const [isEditDatasetMode, setIsEditDatasetMode] = useState(false);
-  const [
-    showUpdateDatasetMetadataModal,
-    setShowUpdateDatasetMetadataModal,
-  ] = useState(false);
+
   const [datasetToEdit, setDatasetToEdit] = useState<Dataset | null>(null);
-  const [datasetMetadataToEdit, setDatasetMetadataToEdit] = useState<{
-    [key: string]: string;
-  } | null>(null);
+  const [datasetMetadataToShow, setDatasetMetadataToShow] = useState<
+    | {
+        [key: string]: string;
+      }
+    | null
+    | undefined
```

Comment:
Why the need for two different representations of "nothing"? Is there a distinction between when this is set to `null` and when it's set to `undefined`?

Reply:
I was trying to adhere to the `Dataset` typescript typing where `dataset_metadata` could be both null and undefined as in not being a property returned in the Dataset response but I think that the property is always returned so I've updated the typing now.